core/src/Streamly/Internal/Data/Stream/Generate.hs

```diff
@@ -140,3 +147,25 @@ fromFoldable = fromStreamK . K.fromFoldable
 {-# INLINE fromFoldableM #-}
 fromFoldableM :: (Monad m, Foldable f) => f (m a) -> Stream m a
 fromFoldableM = Prelude.foldr Stream.consM Stream.nil
+
+-- XXX This should be in the StreamK module and should be named fromPtrN or
+-- fromPtrNUnsafe
+{-# INLINE ptrtoStreamK #-}
+ptrtoStreamK :: forall m a. (MonadIO m, Storable a) =>
+  Ptr a -> Int -> K.Stream m a
+ptrtoStreamK p = D.toStreamK . D.fromPtrN p
+
+-- XXX Should we rename this to fromAddrNUnsafe?
+{-# INLINE fromAddrN# #-}
+fromAddrN# :: (MonadIO m, Storable a) => Int -> Addr# -> Stream m a
+fromAddrN# n addr =
+    -- XXX You probably don't need a castPtr here
+    -- XXX Should look something like the below
+    -- Stream.fromStreamK $ StreamK.fromPtrNUnsafe (Ptr addr) n
+    Stream.fromStreamK $ ptrtoStreamK (Ptr addr) n
+
+-- XXX Should we rename this to fromPtrNUnsafe?
```

Comment:
When you are dealing with Ptrs you know what you are doing. We can add a warning in the docs.